error.go

```diff
@@ -33,6 +32,43 @@ type Err struct {
 	line int
 }
 
+// Locationer is an interface that represents a certain class of errors that
+// contain the location information from where they were raised.
+type Locationer interface {
```

Comment:
Personally, this convention of adding "er" really grates.
YMMV, but I'd prefer "ErrorWithLocation" or literally anything else more descriptive and using correct English instead of a made up word. Maybe it's just me who feels like this about this "er" thing. :-)

It only seems to be used internally so no need to export it.

Reply:
@wallyworld nah I agree, @tlm what you think?

Reply:
It just falls in line with standard go. If we call it ErrorLocation then I think it should also inherit the error interface as we are implying that.

But yes happy to change the name.

Reply:
I see your point now @tlm, the `Locationer` is not necessarily an error, it's just something that provides a method
```go
Location() (function string, line int)
```
I can make it lowercase though.

Reply:
I still think this should be exported by the library.

Reply:
@wallyworld @tlm though maybe we should export `Locationer` as it is referred to in the comments for `NewTimeout`, etc?

Reply:
We definitely should. This is a public interface that exposes useful error functions. No need to keep things wrapped up.

Reply:
Locationer is fine. ErrorWithLocation sounds like a struct that is composed of error.